### ISSUE TITLE
Bert with ner -> run_classifier

### ISSUE THREAD

**Issue by NameBrez** [google-research/bert#672]
Hello,
I've been tackling NER with BERT problem, using the run classifier and custom processor. I've pretrained the model on my language and would like to try something like: In given text I would like o recognize book titles. I've created a training set in this matter: I've splitted the text into words and put them in one collumn in the next colunm I've put the label (O-other, TS-title start, TM-title middle). The problem Im anticipating is the number of occurances of TS and TM in my text as even in a sentence book titles are usualy less than 10% of a sentence thus Im afraid the model wont learn recognizing book titles but rather classify everyting as O and get good accuracy. 

Is this even a way to tacle such problem? I've seen some implementations here but Im wondergin how you've done it as you are mentioning using bert and ner in the article. Is there any samples like the squad one?
Best

**Comment by NameBrez** [comment]
![image](https://user-images.githubusercontent.com/25982352/58818826-1119d000-862f-11e9-9a8a-b8bc4e8ec065.png)
This is what I am looking at

**Comment by Anticsss** [comment]
can you please tell me how you are fine tuning bert for your custom data ?

**Comment by bhavykhatri** [comment]
Yeah I am also stuck at the same point.

**Comment by inessoveral** [comment]
@NameBrez @bhavykhatri @bhavykhatri I am also struggling with this. Have you found a solution?

**Comment by AlxndrMlk** [comment]
@NameBrez:

I think that you might try adding weights to your loss function. In my experience it hugely helps. 

@Anticsss, @bhavykhatri, @inessoveral,

For NER fine-tuning you can refer to these excellent resources for more details:

https://gab41.lab41.org/how-to-fine-tune-bert-for-named-entity-recognition-2257b5e5ce7e
https://gab41.lab41.org/lessons-learned-fine-tuning-bert-for-named-entity-recognition-4022a53c0d90
https://github.com/huggingface/transformers/blob/master/examples/run_ner.py

Hope you'll find them helpful.
